UPGRADING

```diff
@@ -40,13 +40,108 @@ PHP 8.4 UPGRADE NOTES
+  . Gettext:
+    . bind_textdomain_codeset, textdomain and d(*)gettext functions now throw an exception
+      if the domain argument is empty.
+  . Intl:
+    . resourcebundle_get(), ResourceBundle::get(), and accessing offsets on a
+      ResourceBundle object now throw:
+      - TypeError for invalid offset types
+      - ValueError for an empty string
+      - ValueError if the integer index does not fit in a signed 32 bit integer
+    . IntlDateFormatter::__construct() throws a ValueError if the locale is invalid.
+    . NumberFormatter::__construct() throws a ValueError if the locale is invalid.
+  . MBString:
+    . mb_encode_numericentity() and mb_decode_numericentity() now check that
+      the $map is only composed of integers, if not a ValueError is thrown.
+    . mb_http_input() now always throws a ValueError if the $type is invalid.
+    . mb_http_output() now checks that the $encoding parameter does not
+      contain any null bytes. If it does, a ValueError is now thrown.
+  . ODBC:
+    . odbc_fetch_row() returns false when a value less than or equal to 0 is
+      passed for parameter $row. Now, a warning is emitted in this case.
+  . PCNTL:
+    . The functions pcntl_sigprocmask(), pcntl_sigwaitinfo() and
+      pcntl_sigtimedwait() now throw:
+      - A ValueError if the $signals array is empty (except for
+        pcntl_sigprocmask() if the $mode is SIG_SETMASK).
+      - A TypeError if a value of the $signals array is not an integer
+      - A ValueError if a value of the $signals array is not a valid signal number
+    . The function pcntl_sigprocmask() now throw:
+      - A ValueError if $mode is not one of SIG_BLOCK, SIG_UNBLOCK, or SIG_SETMASK
+    . The function pcntl_sigtimedwait() now throw:
+      - A ValueError if $seconds is less than 0
+      - A ValueError if $nanoseconds is less than 0 or greater than 1e9
+      - A ValueError if both $seconds and $nanoseconds are 0
+  . SimpleXML:
+    . Calling simplexml_import_dom() with a non-XML object now throws a
+      TypeError instead of a ValueError.
+  . Standard:
+    . round() now validates the value of the $mode parameter and throws a
+      ValueError for invalid modes. Previously invalid modes would have been
+      interpreted as PHP_ROUND_HALF_UP.
+    . The str_getcsv() function now throws ValueErrors when the $separator and
+      $enclosure arguments are not one byte long, or if the $escape is not one
+      byte long or the empty string. This aligns the behaviour to be identical
+      to that of fputcsv() and fgetcsv().
+    . php_uname() now throws ValueErrors if the $move parameter is invalid.
+    . The "allowed_classes" option for unserialize() now throws TypeErrors and
+      ValueErrors if it is not an array of class names.
+  . XMLReader:
+    . Passing an invalid character encoding to XMLReader::open() or
+      XMLReader::XML() now throws a ValueError.
+    . Passing a string containing NULL bytes previously emitted a
```

Comment:
We should unify this; sometimes we have "null bytes", sometimes "NULL bytes". In my opinion, that should be "NUL bytes", but I'm fine with any other wording.

Reply:
I don't know why use `null bytes` in the PHP ValueError maybe someone else remembers, but I will change the casing.

Reply:
> In my opinion, that should be "NUL bytes"

Agreed.